azurelinuxagent/common/protocol/wire.py

```diff
@@ -1024,22 +1031,58 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
         try:
             ret = host_func()
         except (ResourceGoneError, InvalidContainerError) as e:
+            old_container_id = self.host_plugin.container_id
+            old_role_config_name = self.host_plugin.role_config_name
+
             msg = "Request failed with the current host plugin configuration." \
                   "ContainerId: {0}, role config file: {1}. Fetching new goal state and retrying the call." \
-                  "Error: {2}".format(self.host_plugin.container_id, self.host_plugin.role_config_name, ustr(e))
+                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(e))
             logger.info(msg)
+            add_event(name=AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.HostPlugin,
+                      is_success=False,
+                      message=msg,
+                      log_event=False)
 
             self.update_goal_state(forced=True)
+
+            new_container_id = self.host_plugin.container_id
+            new_role_config_name = self.host_plugin.role_config_name
             msg = "Host plugin reconfigured with new parameters. " \
-                  "ContainerId: {0}, role config file: {1}.".format(self.host_plugin.container_id,
-                                                                    self.host_plugin.role_config_name)
+                  "ContainerId: {0}, role config file: {1}.".format(new_container_id, new_role_config_name)
             logger.info(msg)
 
-            ret = host_func()
-        except Exception:
-            raise
+            try:
+                ret = host_func()
+                if ret:
+                    msg = "Request succeeded using the host plugin channel after goal state refresh." \
+                          "ContainerId changed from {0} to {1}, " \
+                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id,
+                                                                             old_role_config_name, new_role_config_name)
+                    add_event(name=AGENT_NAME,
+                              version=CURRENT_VERSION,
+                              op=WALAEventOperation.HostPlugin,
+                              is_success=True,
+                              message=msg,
+                              log_event=False)
+
+            except (ResourceGoneError, InvalidContainerError) as e:
+                add_event(name=AGENT_NAME,
+                          version=CURRENT_VERSION,
+                          op=WALAEventOperation.HostPlugin,
+                          is_success=False,
+                          message=ustr(e),
```

Comment:
The message should be more explicit, something along the lines of the message at line 1059.
Also, probably we should not include the entire exception, but only its type (in case there are any secrets in the request)

Reply:
Good catch, thanks. Updated.